docs/how-to-guides/shared-libraries.rst

```diff
@@ -0,0 +1,250 @@
+.. SPDX-FileCopyrightText: 2024 The meson-python developers
+..
+.. SPDX-License-Identifier: MIT
+
+.. _shared-libraries:
+
+**********************
+Using shared libraries
+**********************
+
+Python projects may build shared libraries as part of their project, or link
+with shared libraries from a dependency. This tends to be a common source of
+issues, hence this page aims to explain how to include shared libraries in
+wheels, any limitations and gotchas, and how support is implemented in
+``meson-python`` under the hood.
+
+We distinguish between *internal* shared libraries that are built as part of
+the project, and *external* shared libraries that are provided by project
+dependencies and that are linked with the project build artifacts.
+For internal shared libraries, we also distinguish whether the shared library
+is being installed to its default system location (typically
+``/usr/local/lib`` on Unix-like systems, and ``C:\\lib`` on Windows - we call
+this ``libdir`` in this guide) or to a location in ``site-packages`` within the
+Python package install tree. All these scenarios are (or will be) supported,
+with some caveats:
+
++-----------------------+------------------+---------+-------+-------+
+| shared library source | install location | Windows | macOS | Linux |
++=======================+==================+=========+=======+=======+
+| internal              | libdir           | no (1)  | ✓     | ✓     |
```

Comment:
I think there is a very serious lack of use of emojis in our documentation 😄. I was thinking to use emojis for the check mark columns of this table ✔️ and 🚧 maybe

Reply:
I gave that a try, looks reasonable. The ✔️ didn't work because it doesn't play well with the dark theme, but 🗹 did.

Kept it in a separate commit so it's easily reversible. Messing with table formatting is a bit fiddly.

Reply:
It looks fine with the GitHub dark mode. Is it an issue with the Sphynx theme? This was just a silly suggestion, the table looks fine as it was without the emojis and we can keep it like that if it makes editing it cumbersome.

Reply:
Probably yes. Just that emoji didn't seem to change color, while the other one did.

I'm happy to keep the commit, it's slightly nicer I'd say. Now that I've manually realigned the table it should be fine.

Reply:
I build the docs locally and on macOS 12.7 the unicode character you choose is not shown correctly. It looks like the glyph is missing in the font used by the Sphinx theme. Or something like this. I'm sure we can fix it, somehow, but I'm not sure it is worth the effort. Maybe we should go back to the simpler symbols in the table, at least until someone does not figure out what's wrong with emojis and the Furo theme.

Reply:
> Maybe we should go back to the simpler symbols in the table,

Agreed, let me drop the last commit.